dlt/common/runners/pool_runner.py

```diff
@@ -38,10 +40,102 @@ def submit(self, fn: Callable[P, T], /, *args: P.args, **kwargs: P.kwargs) -> Fu
+            thread_name_prefix: An optional name prefix to give our threads.
+            initializer: A callable used to initialize worker threads.
+            initargs: A tuple of arguments to pass to the initializer.
+        """
+        super().__init__(
+            max_workers,
+            thread_name_prefix=thread_name_prefix,
+            initializer=initializer,
+            initargs=initargs,
+        )
+        self.timeout = timeout
+        self._is_alive: bool = None  # set on shutdown
+
+    def shutdown(self, wait: bool = True, *, cancel_futures: bool = False) -> None:
+        if not wait or self.timeout is None:
+            # behave exactly like the std-lib version
+            return super().shutdown(wait=wait, cancel_futures=cancel_futures)
+
+        # initiate shutdown but return immediately
+        super().shutdown(wait=False, cancel_futures=cancel_futures)
+
+        deadline = time.monotonic() + self.timeout
+        for t in list(self._threads):
+            remaining = deadline - time.monotonic()
+            if remaining <= 0:
+                break
+            t.join(remaining)
+
+        alive = [t for t in self._threads if t.is_alive()]
+        if alive:
+            logger.error(
+                f"Shutdown of dlt thread pool {self._thread_name_prefix} could not be completed in"
+                f" {self.timeout} seconds. All the tasks in the pool are completed, nevertheless"
+                " some threads didn't join. This may happen, for example, if thread local storage "
+                " is used with object with __del__ method that locks thread when garbage"
+                " collecting or if os.fork() is used. NOTE: if you use database clients or other"
+                " Python libraries in resource or custom destinations, they can do that without"
+                " you knowing. WARNING! This process will still try to join those threads on exit"
+                " and most probably will lock indefinitely."
+            )
+            logger.info(
+                f"Shutdown of dlt thread pool {self._thread_name_prefix} - "
+                f"{len(alive)} thread(s) still alive: {', '.join(t.name for t in alive)}."
+            )
+        self._is_alive = len(alive) > 0
+
+
+def get_default_start_method(method_: str) -> str:
+    """Sets method to `spawn` is running in one of orchestrator tasks.
+
+    Called when explicit start method is not set on `PoolRunnerConfiguration`
+    """
+    if method_ == "fork":
+        from dlt.common.runtime.exec_info import (
+            is_running_in_airflow_task,
+            is_running_in_dagster_task,
+            is_running_in_prefect_flow,
+        )
+
+        for m_ in [
+            is_running_in_airflow_task,
+            is_running_in_dagster_task,
+            is_running_in_prefect_flow,
+        ]:
+            if m_():
+                logger.info(
+                    f"Switching pool start method to `spawn` because `{m_.__name__}` is True"
```

Comment:
praise: nice way of using the functions name. very concise. doesnt look like a human wrote it to me, but actually very nice.